helix-core/src/syntax.rs

```diff
@@ -2456,6 +2456,44 @@ pub fn pretty_print_tree<W: fmt::Write>(fmt: &mut W, node: Node) -> fmt::Result
     }
 }
 
+pub fn get_highlight_for_node_at_position(
+    syntax: Option<&Syntax>,
+    text: RopeSlice,
+    pos: usize,
+) -> Option<Highlight> {
+    let mut highlight: Option<Highlight> = None;
+    if let Some(syntax) = syntax {
+        let pos = text.char_to_byte(pos);
+
+        let node: Node = match syntax
+            .tree()
+            .root_node()
+            .descendant_for_byte_range(pos, pos)
+        {
+            Some(node) => node,
+            None => return None,
+        };
+
+        for event in syntax
+            .highlight_iter(text.slice(..), Some(node.byte_range()), None)
```

Comment:
The `range` argument here should be the same as we use [in rendering](https://github.com/helix-editor/helix/blob/61814fea7f779267bd0c6bc9428c1f53142ee7f1/helix-term/src/ui/editor.rs#L295-L304) so that the command always returns the highlight as you see from syntax highlighting